firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -569,7 +570,9 @@ private void createBundleCache() {
                   + "bundle_id TEXT PRIMARY KEY, "
                   + "create_time_seconds INTEGER, "
                   + "create_time_nanos INTEGER, "
-                  + "schema_version INTEGER)");
+                  + "schema_version INTEGER, "
+                  + "total_documents INTEGER, "
+                  + "total_bytes INTEGER)");
```

Comment:
total_documents and total_bytes are only used during loading, they are not of much use after it is loaded, and we should not persist them.

Reply:
I have added and removed these fields quite a few times now. Unfortunately, it seems easier to just include them in BundleMetadata, as the combined state can then be passed around in one object. Note that unlike JS, the BundleLoader cannot just grab these values from the JSON stream since they need to be deserialized first. If we want to omit these values from persistence, I think they are only two strategies:

- We create a BundleMetadata and a BundleMetadataDuringLoading and only use BundleMetadata in Storage
- We don't write these values to persistence and use -1 or similar values to populate BundleMetadata when we deserialize

Both of these options seems much less clean to me.

Reply:
SG.